File: hadoop-yarn-spark/scala/Scala-Spark/src/main/scala/TestRecord.scala
class TestRecord {

  private var logicalName = ""
  private var duration = 0L
  private var occurrences = 0

  def this(logicalName: String) {
    this()
    this.logicalName = logicalName
  }

  def getLogicalName: String = logicalName

  def addDuration(duration: Long): Unit = {
    this.duration += duration
    this.occurrences += 1
  }

  def getDuration: Long = duration

  def getOccurrences: Integer = occurrences

}

File: hadoop-yarn-spark/java/java-spark/src/main/java/com/maaksoft/spark/TestRecord.java
package com.maaksoft.spark;

public class TestRecord {

    private String logicalName;
    private Long duration = 0L;
    private Integer occurrences = 0;

    public TestRecord(String logicalName) {
        this.logicalName = logicalName;
    }

    public String getLogicalName() {
        return logicalName;
    }

    public void addDuration(Long duration) {
        this.duration += duration;
        this.occurrences++;
    }

    public Long getDuration() {
        return duration;
    }

    public Integer getOccurrences() {
        return occurrences;
    }

}
